lib/transmuxer/h265.js

```diff
@@ -0,0 +1,855 @@
+    let topOffset = 0;
+    let bottomOffset = 0;
+
+    // SPS
+    gb.readBits(4); // video_paramter_set_id
+    const maxSubLayersMinus1 = gb.readBits(3);
+    gb.readBoolean(); // temporal_id_nesting_flag
+
+    // profile_tier_level begin
+    const generalProfileSpace = gb.readBits(2);
+    const generalTierFlag = gb.readBits(1);
+    const generalProfileIdc = gb.readBits(5);
+    const generalProfileCompatibilityFlags1 = gb.readUnsignedByte();
+    const generalProfileCompatibilityFlags2 = gb.readUnsignedByte();
+    const generalProfileCompatibilityFlags3 = gb.readUnsignedByte();
+    const generalProfileCompatibilityFlags4 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags1 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags2 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags3 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags4 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags5 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags6 = gb.readUnsignedByte();
+    const generalLevelIdc = gb.readUnsignedByte();
+    const subLayerProfilePresentFlag = [];
+    const subLayerLevelPresentFlag = [];
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      subLayerProfilePresentFlag.push(gb.readBoolean());
+      subLayerLevelPresentFlag.push(gb.readBoolean());
+    }
+    if (maxSubLayersMinus1 > 0) {
+      for (let i = maxSubLayersMinus1; i < 8; i++) {
+        gb.readBits(2);
+      }
+    }
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      if (subLayerProfilePresentFlag[i]) {
+        gb.readBits(88);
+      }
+      if (subLayerLevelPresentFlag[i]) {
+        gb.readUnsignedByte();
+      }
+    }
+    // profile_tier_level end
+
+    gb.readUnsignedExpGolomb(); // seq_parameter_set_id
+    const chromaFormatIdc = gb.readUnsignedExpGolomb();
+    if (chromaFormatIdc == 3) {
+      gb.readBits(1); // separate_colour_plane_flag
+    }
+    const picWidthInLumaSamples = gb.readUnsignedExpGolomb();
+    const picHeightInLumaSamples = gb.readUnsignedExpGolomb();
+    const conformanceWindowFlag = gb.readBoolean();
+    if (conformanceWindowFlag) {
+      leftOffset += gb.readUnsignedExpGolomb();
+      rightOffset += gb.readUnsignedExpGolomb();
+      topOffset += gb.readUnsignedExpGolomb();
+      bottomOffset += gb.readUnsignedExpGolomb();
+    }
+    const bitDepthLumaMinus8 = gb.readUnsignedExpGolomb();
+    const bitDepthChromaMinus8 = gb.readUnsignedExpGolomb();
+    const log2MaxPicOrderCntLsbMinus4 = gb.readUnsignedExpGolomb();
+    const subLayerOrderingInfoPresentFlag = gb.readBoolean();
+    for (
+      let i = subLayerOrderingInfoPresentFlag ? 0 : maxSubLayersMinus1;
+      i <= maxSubLayersMinus1;
+      i++
+    ) {
+      gb.readUnsignedExpGolomb(); // max_dec_pic_buffering_minus1[i]
+      gb.readUnsignedExpGolomb(); // max_num_reorder_pics[i]
+      gb.readUnsignedExpGolomb(); // max_latency_increase_plus1[i]
+    }
```

Comment:
If I'm reading this correctly, when subLayerOrderingInfoPresentFlag is false, we do this loop once.  `i` begins at `maxSubLayersMinus1`, and we loop while `i <= maxSubLayersMinus1`.  Is that correct?

If so, I think it would be easier to understand if we split it up:

```js
    if (subLayerOrderingInfoPresentFlag) {
      // Skip each layer
      for (let i = 0; i <= maxSubLayersMinus1; i++) {
        gb.readUnsignedExpGolomb(); // max_dec_pic_buffering_minus1[i]
        gb.readUnsignedExpGolomb(); // max_num_reorder_pics[i]
        gb.readUnsignedExpGolomb(); // max_latency_increase_plus1[i]
      }
    } else {
      // Skip one layer
      gb.readUnsignedExpGolomb(); // max_dec_pic_buffering_minus1[i]
      gb.readUnsignedExpGolomb(); // max_num_reorder_pics[i]
      gb.readUnsignedExpGolomb(); // max_latency_increase_plus1[i]
    }
```

If you don't like repeating yourself, you could also move the read calls to a method.

Reply:
Done!